src/librustdoc/clean/utils.rs

```diff
@@ -335,12 +335,12 @@ crate fn print_evaluated_const(tcx: TyCtxt<'_>, def_id: DefId) -> Option<String>
 fn format_integer_with_underscore_sep(num: &str) -> String {
     let num_chars: Vec<_> = num.chars().collect();
     let mut num_start_index = if num_chars.get(0) == Some(&'-') { 1 } else { 0 };
-    let chunk_size = match (num_chars.get(num_start_index), num_chars.get(num_start_index + 1)) {
-        (Some('0'), Some('b' | 'x')) => {
+    let chunk_size = match num.as_bytes() {
+        [b'0', b'b', ..] => {
```

Comment:
Oh oops, I suggested a change in behavior,  sorry.
```suggestion
        [b'0', b'b' | b'x', ..] => {
```

Reply:
ced3c5ad26857e2c426306d821641e91dc8e860d